judge/contest_format/ioi16.py

```diff
@@ -0,0 +1,98 @@
+        score = 0
+        format_data = {}
+
+        with connection.cursor() as cursor:
+            cursor.execute('''
+                SELECT *
+                FROM (
+                         SELECT q.prob,
+                                q.date,
+                                q.batch_points
+                         FROM (
+                                  SELECT cp.id          as `prob`,
+                                         sub.id         as `subid`,
+                                         sub.date       as `date`,
+                                         tc.points      as `points`,
+                                         tc.batch       as `batch`,
+                                         Min(tc.points) as `batch_points`
+                                  FROM judge_contestproblem cp
+                                           INNER JOIN
+                                       judge_contestsubmission cs
+                                       ON (cs.problem_id = cp.id AND cs.participation_id = %s)
+                                           LEFT OUTER JOIN
+                                       judge_submission sub ON (sub.id = cs.submission_id)
+                                           INNER JOIN judge_submissiontestcase tc
+                                                      ON sub.id = tc.submission_id
+                                  GROUP BY cp.id, tc.batch, sub.id
+                              ) q
+                                  INNER JOIN (
+                             SELECT prob, batch, MAX(r.batch_points) max_batch_points
+                             FROM (
+                                      SELECT cp.id          as `prob`,
+                                             tc.batch       as `batch`,
+                                             Min(tc.points) as `batch_points`
+                                      FROM judge_contestproblem cp
+                                               INNER JOIN
+                                           judge_contestsubmission cs
+                                           ON (cs.problem_id = cp.id AND cs.participation_id = %s)
+                                               INNER JOIN
+                                           judge_submission sub ON (sub.id = cs.submission_id)
+                                               INNER JOIN judge_submissiontestcase tc
+                                                          ON sub.id = tc.submission_id
+                                      GROUP BY cp.id, tc.batch, sub.id
+                                  ) r
+                             GROUP BY prob, batch
+                         ) p
+                                       ON p.prob = q.prob AND p.batch = q.batch
+                         WHERE p.max_batch_points = q.batch_points
+                           AND p.prob = q.prob
+                         GROUP BY q.prob, q.batch
+                     ) best
+                ORDER BY best.date;
```

Comment:
Do we need to order by `best.date`? From what I can see, we can find `max(dt)` in the Python code instead (and it might make the intention more clear to others).

Reply:
Sure.
```suggestion
```